relayer/src/tx/connection.rs

```diff
@@ -231,3 +296,164 @@ pub fn build_conn_try_and_send(opts: ConnectionOpenTryOptions) -> Result<String,
 
     Ok(dest_chain.send(dest_msgs, key, "".to_string(), 0)?)
 }
+
+/// Attempts to send a MsgConnOpenAck to the dest_chain.
+pub fn build_conn_ack(
+    dest_chain: &mut CosmosSDKChain,
+    src_chain: &CosmosSDKChain,
+    opts: &ConnectionOpenOptions,
+) -> Result<Vec<Any>, Error> {
+    let _expected_dest_connection =
+        validated_expected_connection(dest_chain, src_chain, ConnectionMsgType::OpenAck, opts)
+            .map_err(|e| {
+                Kind::ConnOpenAck(
+                    opts.src_connection_id.clone(),
+                    "ack options inconsistent with existing connection on destination chain"
+                        .to_string(),
+                )
+                .context(e)
+            })?;
+
+    let src_connection = src_chain
+        .query_connection(&opts.src_connection_id.clone(), ICSHeight::default())
+        .map_err(|e| {
+            Kind::ConnOpenAck(
+                opts.src_connection_id.clone(),
+                "missing connection on source chain".to_string(),
+            )
+            .context(e)
+        })?;
+
+    // TODO - check that the src connection is consistent with the ack options
+
+    // TODO - Build add **send** the message(s) for updating client on source (when we don't need the key seed anymore)
+    // TODO - add check if it is required
+    // let (key, signer) = src_chain.key_and_signer(&opts.src_signer_seed)?;
+    // build_update_client_and_send(ClientOptions {
```

Comment:
So currently we're not updating the client on the source chain (which is fine). I see a bit of potential for factoring out the common parts of `build_conn_ack` `build_conn_confirm`, but we'll handle all these later!

Reply:
Yeah, i am waiting for a PR from Andy to get rid of the seed_files and will do the src update client at that time. Right now during testing I do it in a separate CLI to make sure the client on source has a recent state.